gwcs/coordinate_frames.py

```diff
@@ -399,14 +450,36 @@ class TemporalFrame(CoordinateFrame):
     """
 
     def __init__(self, axes_order=(0,), reference_time=None,
-                 reference_frame=astropy.time.Time, unit=None,
+                 reference_frame=None, unit=None,
                  axes_names=None, name=None, axis_physical_types=None):
+        if reference_frame is None:
+            reference_frame = {}
+        elif not isinstance(reference_frame, dict):
+            raise TypeError("reference_frame must be a dict.")
 
         super().__init__(naxes=1, axes_type="TIME", axes_order=axes_order,
                          axes_names=axes_names, reference_frame=reference_frame,
                          unit=unit, name=name,
                          reference_position=reference_time, axis_physical_types=axis_physical_types)
 
+    @property
+    def _world_axis_object_classes(self):
+        if self.reference_position:
+            return {'temporal': (
+                RelativeTime,
```

Comment:
As @astrofrog suggested, doing this allows `issubclass(..., Time)` to work if people want to type check this.